### ISSUE TITLE
WithAgentHealthProber(...) should pass ManagedClusterAddon to correctly detect agent namespace

### ISSUE THREAD

**Issue by tamalsaha** [open-cluster-management-io/addon-framework#221]
Currently `WithAgentHealthProber()` does not have access to the agent namespace. This is required for workapi type prober to correctly identify the workloads for a given cluster.

**Comment by qiujian16** [comment]
@zhiweiyin318 I think with the recent change on addon-framework, this has been resolved? Would you help to check?

**Comment by zhiweiyin318** [comment]
yes, we have added a new func in WorkHealthProber which can pass addon .  
https://github.com/open-cluster-management-io/addon-framework/blob/57b1d2859f501d3cbb05143f92c535f7e07f189b/pkg/agent/inteface.go#L172
@tamalsaha  please try this new func.
